### DISCUSSION TITLE
Access an element within an array of arrays

### DISCUSSION THREAD

**Discussion by glcash57** [nlohmann/json#3115] [Q&A] [answered]
version 3.9.1

char text[] = R"(
{
  "Object": {
      "array": [5, 10, 15, 20],
      "array_of_arrays": [[5,10,15], [20,25,30]]
  }
}
)";

json jp = json::parse(text); 

int ids0 = jp["Object"]["array"][0];               // works for element 0 in array

int ids00 = jp["Object"]["array_of_arrays"][0][0]; // does not work for element 0,0 in array of arrays

How can I access these elements?

**Comment by gregmarr** [comment]
What does "does not work" mean?  Do you get an error message?

**Reply by glcash57** [reply]
I'm sorry, I am a newbie to this. How would I get an error message?

**Reply by glcash57** [reply]
Figured out how to get error messages:
 json jp;
    try
    {
        // parsing input with a syntax error
       jp = json::parse(text);
    }
    catch (json::parse_error& e)
    {
        // output exception information
        std::cout << "message: " << e.what() << '\n'
            << "exception id: " << e.id << '\n'
            << "byte position of error: " << e.byte << std::endl;
    }

not getting an error message

**Reply by gregmarr** [reply]
That's only for parsing.  You said that the `ids0` line works but the `ids00` line doesn't work.  How are you determining that it doesn't work?

It's working just fine for me here: https://godbolt.org/z/1zqeo7eK7

**Reply by glcash57** [reply]
I am truly sorry to have bothered you. It was a case issue that I should have caught. In my original code I was using upper-case "Object" when the original json used lower-case "object".

Thanks again for your patient help!

**Reply by gregmarr** [reply]
@glcash57 Can you close this issue then?

**Reply by glcash57** [reply]
Sorry, I would like to, but I cannot find a button to do so. Where is it?

**Reply by nlohmann** [reply]
It's fine - you marked an answer which is all you can do in GitHub Discussions.

**Reply by gregmarr** [reply]
Ah, I didn't realize that.  Thanks @nlohmann.

**Comment by nlohmann** [comment, marked as answer]
In order to asses your issue, we need the following information:

- What is the issue you have?

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

- What is the expected behavior?

- And what is the actual behavior instead?

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

- Did you use a released version of the library or the version from the `develop` branch?

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?
